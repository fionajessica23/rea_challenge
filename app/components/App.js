import React from 'react';
import PropTypes from 'prop-types';
import PropertyPanel from './PropertyPanel';
import './App.scss';
import * as actionCreators from './actionCreators';
import { connect } from 'react-redux';

function App(props) {
  const { listings, addSavedListing, removeSavedListing } = props;

  const results = listings.filter(listing => listing.isSaved === false);
  const saved = listings.filter(listing => listing.isSaved === true);

  return (
    <div className="container">
      <h1 className="header">For Sale</h1>
      <div className="row">
        <PropertyPanel
          headingText="Results"
          results={results}
          handlePropBtnClick={addSavedListing}
          buttonText="add property"
        />
        <PropertyPanel
          headingText="Saved Properties"
          results={saved}
          handlePropBtnClick={removeSavedListing}
          buttonText="remove property"
        />
      </div>
    </div>
  );
}

App.propTypes = {
  listings: PropTypes.arrayOf(PropTypes.shape({
    price: PropTypes.string.isRequired,
    agency: PropTypes.shape({
      brandingColors: PropTypes.shape({
        primary: PropTypes.string.isRequired,
      }).isRequired,
      logo: PropTypes.string.isRequired,
    }).isRequired,
    id: PropTypes.string.isRequired,
    mainImage: PropTypes.string.isRequired,
  })).isRequired,
  addSavedListing: PropTypes.func.isRequired,
  removeSavedListing: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  listings: state.listings,
});

const mapActionsToProps = {
  addSavedListing: actionCreators.addSavedListing,
  removeSavedListing: actionCreators.removeSavedListing,
};

export default connect(mapStateToProps, mapActionsToProps)(App);
