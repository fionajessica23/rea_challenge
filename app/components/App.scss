@import "./style-variables.scss";

html {
  margin: 0;
  font-size: $root-font;
  background-color: $anti-flash-white;
  font-family: Helvetica, sans-serif;
}

.header {
  padding: $med-space 0;
  text-align: center;
  font-size: $xxlarge-font;
}

.panelsWrapper {
  @media (min-width: $med-breakpoint) {
    display: flex;
    justify-content: space-evenly;
  }
}
