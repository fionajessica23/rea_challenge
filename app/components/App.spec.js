import React from 'react';
import renderer from 'react-test-renderer';
import { App } from './App';

jest.mock('./PropertyPanel.js', () => 'PropertyPanel');

const DEFAULT_PROPS = {
  listings: [
    {
      price: '$650,000',
      agency: {
        brandingColors: {
          primary: '#000000',
        },
        logo: 'http://i2.au.reastatic.net/agencylogo/WVYSSK/2/20140701084436.gif',
      },
      id: '9',
      mainImage: 'http://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg',
    },
    {
      price: '$300,000',
      agency: {
        brandingColors: {
          primary: '#111111',
        },
        logo: 'http://i2.au.reastatic.net/agencylogo/WVYSSK/2/20140701084436.gif',
      },
      id: '5',
      mainImage: 'http://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg',
    },
  ],
  savedIds: ['5'],
  addSavedListing: () => { },
  removeSavedListing: () => { },
};

it('renders App with listings', () => {
  const tree = renderer
    .create(<App {...DEFAULT_PROPS} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders App with saved properties id equal to 9', () => {
  const propWithSavedIds = { ...DEFAULT_PROPS, savedIds: ['9'] };
  const tree = renderer
    .create(<App {...propWithSavedIds} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders App with no saved listings', () => {
  const propWithNoSavedIds = { ...DEFAULT_PROPS, savedIds: [] };
  const tree = renderer
    .create(<App {...propWithNoSavedIds} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
