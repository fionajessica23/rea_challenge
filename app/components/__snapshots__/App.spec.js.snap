// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App with listings 1`] = `
<div>
  <h1
    className="mockClassName-header"
  >
    For Sale
  </h1>
  <div
    className="mockClassName-panelsWrapper"
  >
    <PropertyPanel
      buttonText="save"
      handleClick={[Function]}
      headingText="Available"
      listings={
        Array [
          Object {
            "agency": Object {
              "brandingColors": Object {
                "primary": "#000000",
              },
              "logo": "http://i2.au.reastatic.net/agencylogo/WVYSSK/2/20140701084436.gif",
            },
            "id": "9",
            "mainImage": "http://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg",
            "price": "$650,000",
          },
          Object {
            "agency": Object {
              "brandingColors": Object {
                "primary": "#111111",
              },
              "logo": "http://i2.au.reastatic.net/agencylogo/WVYSSK/2/20140701084436.gif",
            },
            "id": "5",
            "mainImage": "http://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg",
            "price": "$300,000",
          },
        ]
      }
    />
    <PropertyPanel
      buttonText="remove"
      handleClick={[Function]}
      headingText="Saved Properties"
      listings={
        Array [
          Object {
            "agency": Object {
              "brandingColors": Object {
                "primary": "#111111",
              },
              "logo": "http://i2.au.reastatic.net/agencylogo/WVYSSK/2/20140701084436.gif",
            },
            "id": "5",
            "mainImage": "http://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg",
            "price": "$300,000",
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`renders App with no saved listings 1`] = `
<div>
  <h1
    className="mockClassName-header"
  >
    For Sale
  </h1>
  <div
    className="mockClassName-panelsWrapper"
  >
    <PropertyPanel
      buttonText="save"
      handleClick={[Function]}
      headingText="Available"
      listings={
        Array [
          Object {
            "agency": Object {
              "brandingColors": Object {
                "primary": "#000000",
              },
              "logo": "http://i2.au.reastatic.net/agencylogo/WVYSSK/2/20140701084436.gif",
            },
            "id": "9",
            "mainImage": "http://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg",
            "price": "$650,000",
          },
          Object {
            "agency": Object {
              "brandingColors": Object {
                "primary": "#111111",
              },
              "logo": "http://i2.au.reastatic.net/agencylogo/WVYSSK/2/20140701084436.gif",
            },
            "id": "5",
            "mainImage": "http://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg",
            "price": "$300,000",
          },
        ]
      }
    />
    <PropertyPanel
      buttonText="remove"
      handleClick={[Function]}
      headingText="Saved Properties"
      listings={Array []}
    />
  </div>
</div>
`;

exports[`renders App with saved properties id equal to 9 1`] = `
<div>
  <h1
    className="mockClassName-header"
  >
    For Sale
  </h1>
  <div
    className="mockClassName-panelsWrapper"
  >
    <PropertyPanel
      buttonText="save"
      handleClick={[Function]}
      headingText="Available"
      listings={
        Array [
          Object {
            "agency": Object {
              "brandingColors": Object {
                "primary": "#000000",
              },
              "logo": "http://i2.au.reastatic.net/agencylogo/WVYSSK/2/20140701084436.gif",
            },
            "id": "9",
            "mainImage": "http://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg",
            "price": "$650,000",
          },
          Object {
            "agency": Object {
              "brandingColors": Object {
                "primary": "#111111",
              },
              "logo": "http://i2.au.reastatic.net/agencylogo/WVYSSK/2/20140701084436.gif",
            },
            "id": "5",
            "mainImage": "http://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg",
            "price": "$300,000",
          },
        ]
      }
    />
    <PropertyPanel
      buttonText="remove"
      handleClick={[Function]}
      headingText="Saved Properties"
      listings={
        Array [
          Object {
            "agency": Object {
              "brandingColors": Object {
                "primary": "#000000",
              },
              "logo": "http://i2.au.reastatic.net/agencylogo/WVYSSK/2/20140701084436.gif",
            },
            "id": "9",
            "mainImage": "http://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg",
            "price": "$650,000",
          },
        ]
      }
    />
  </div>
</div>
`;
